runtime/src/message_processor.rs

```diff
@@ -140,6 +141,7 @@ impl MessageProcessor {
         program_accounts: &mut [&mut Account],
     ) -> Result<(), InstructionError> {
         let program_id = instruction.program_id(&message.account_keys);
+        let is_loader = executable_accounts.len() > 1;
```

Comment:
This is probably the only interesting part of this PR.  I'm not sure if there's a better way to signal that we're about to invoke a loader or what to do about loaders of loaders.